pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -251,6 +253,12 @@ func (f *Fit) EventsToRegister() []framework.ClusterEvent {
 // Checks if a node has sufficient resources, such as cpu, memory, gpu, opaque int resources etc to run a pod.
 // It returns a list of insufficient resources, if empty, then the node has all the resources requested by the pod.
 func (f *Fit) Filter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status {
+	if !f.enableSidecarContainers && hasRestartableInitContainer(pod) {
+		// To avoid the inconsistency in resource calculation between the scheduler
+		// and the older (before v1.28) kubelet
+		return framework.NewStatus(framework.UnschedulableAndUnresolvable, "SidecarContainers feature is disabled")
```

Comment:
`Pod has a restartaable init container and the SidecarContainers feature is disabled - cannot schedule this pod` (if this message is going to a human)